source/core/rheological_model.cc

```diff
@@ -1,33 +1,26 @@
 #include <core/rheological_model.h>
 
 std::shared_ptr<RheologicalModel>
-RheologicalModel::model_cast(
-  const Parameters::PhysicalProperties &physical_properties)
+RheologicalModel::model_cast(const Parameters::Fluid &fluid_properties)
 {
-  if (!physical_properties.fluids[0].non_newtonian_flow)
-    return std::make_shared<Newtonian>(physical_properties.fluids[0].viscosity);
-  else if (physical_properties.fluids[0].non_newtonian_parameters.model ==
-           Parameters::NonNewtonian::Model::powerlaw)
+  if (!fluid_properties.non_newtonian_flow)
```

Comment:
I like that `physical_properties.fluid[0]` is now `fluid_properties`, but I see less clearly how this will apply with two-fluids simulations, with one newtonian fluid and the other non-newtonian. Could you clarify how this will be generalized in the future?

Reply:
So the way I see this in when using the VOF method, you will have a fluids vector containing (two or more) fluids, each containing their parameters. Let's say the first fluid, i.e. fluids[0], has its member non_newtonian_flow set to false, and the second fluid has its member non_newtonian_flow set to true. Now instead of using physical_properties, which before contained the properties for all fluids, you now access to each fluid properties through fluid_properties. when generating the rheological_model object for the first fluid, this method will cast the rheological_model object to be a Newtonian model. For the second fluid, it's rheological_model object will be of class Carreau or PowerLaw. This way, when calculating the visocsity in the first fluid, this fluid's rheological_model object will return a constant viscosity. When calculating the viscosity in the second fluid, the returned viscosity will be calculated depending on the rheological model used. I'm not sure if I answered you question correctly, don't hesitate to reiterate

Reply:
yes that clarifies it perfectly, thank you @CaroleAnneDaunais :)

Reply:
no worries :)